### PR TITLE
Add analysis for drainage areas

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

This takes an input AoI, finds its centroid, finds a HUC-12 that contains the centroid, and runs GWLF-E on it. Ultimately, we will scale the GWLF-E results to the input AoI, but for that we need the Drexel / ANS API. So this is not complete until that API has been integrated.

Connects #3553 

## Testing Instructions

* Check out this branch
* Go to http://localhost:8000/ and draw any drainage area
* Download the resulting AoI as GeoJSON
    image
* Copy your API token
* Exercise the new API, like so:
    ```console
    http :8000/api/analyze/drainage-area/ Authorization:"Token $YOUR_TOKEN" < $DOWNLOADED_FILE.geojson
    ```
  - [x] Ensure the call succeeds
  - [x] Ensure the job results are GWLF-E outputs of Loads and monthlies

### CODE REVIEW

**Review by emilyhu0106 — APPROVED** [review]
I tested both stream-based and point-based areas with different sizes, all works well! The changes looks good!

File: src/mmw/apps/geoprocessing_api/views.py [hunk 1]
```diff
@@ -1344,6 +1344,319 @@ def start_analyze_drb_2100_land(request, key=None, format=None):
...
+@swagger_auto_schema(method='post',
+                     request_body=schemas.ANALYZE_REQUEST,
+                     responses={200: schemas.JOB_STARTED_RESPONSE})
+@decorators.api_view(['POST'])
+@decorators.authentication_classes((SessionAuthentication,
+                                    TokenAuthentication, ))
+@decorators.permission_classes((IsAuthenticated, ))
+@decorators.throttle_classes([BurstRateThrottle, SustainedRateThrottle])
+@log_request
+def start_analyze_drainage_area(request, format=None):
```
Comment (emilyhu0106) [inline src/mmw/apps/geoprocessing_api/views.py:1356]: This doesn't seem to be available in swagger right now. Are we going to add it later?
image

Comment (rajadain) [reply src/mmw/apps/geoprocessing_api/views.py:1356]: It is available in the `/analyze/` section:

image

**Comment by rajadain** [comment]
Thanks for reviewing! Merging this now.

**Pull request merged**